### PR TITLE
Add NNAPI tutorial

### PR DESCRIPTION
**Pull request by dreiss**

### CODE REVIEW

File: prototype_source/nnapi_mobilenetv2.rst [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+.. code:: python
+
+    #!/usr/bin/env python
+    import sys
+    import os
+    import torch
+    import torch.utils.bundled_inputs
+    import torch.utils.mobile_optimizer
+    import torch.backends._nnapi.prepare
```
Comment (jeffxtang) [inline prototype_source/nnapi_mobilenetv2.rst:41]: Which PyTorch and torchvision versions are required to run this? I built and installed PyTorch from the master branch on Oct 22, got an error "ModuleNotFoundError: No module named 'torch.backends._nnapi'". Then I built from the latest pytorch master branch, and got a new error when running the script:

Exception: Unsupported node kind ('aten::size') in node %13 : int = aten::size(%input.88, %32) # /Users/jeffxtang/opt/anaconda3/lib/python3.8/site-packages/torchvision/models/mobilenet.py:166:0

I then installed torchvision from the latest source but the script still picks the older 0.7...

**Review by jeffxtang — COMMENTED** [review]
Half way thru the PR and got errors when running the script. The right torch and torchvision versions and setup info would be helpful. Thanks

File: prototype_source/nnapi_mobilenetv2.rst [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+------------------
+
+Now that the models are ready, we can benchmark them on our Android devices.
+See `our performance recipe <https://pytorch.org/tutorials/recipes/mobile_perf.html#android-benchmarking-setup>`_ for details.
+The best-performing models are likely to be the "fully-quantized" models:
+``mobilenetv2-quant_full-cpu.pt`` and ``mobilenetv2-quant_full-nnapi.pt``.
+
+Because these models have bundled inputs, we can run the benchmark as follows:
+
+.. code:: shell
```
Comment (jeffxtang) [inline prototype_source/nnapi_mobilenetv2.rst:148]: The code below doesn't show in the preview, maybe due to some format issue. code => code-block?

File: prototype_source/nnapi_mobilenetv2.rst [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+        for quantize_mode in ["none", "core", "full"]:
+            make_mobilenetv2_nnapi(Path(os.environ["HOME"]) / "mobilenetv2-nnapi", quantize_mode)
+
+
+Running Benchmarks
+------------------
+
+Now that the models are ready, we can benchmark them on our Android devices.
+See `our performance recipe <https://pytorch.org/tutorials/recipes/mobile_perf.html#android-benchmarking-setup>`_ for details.
+The best-performing models are likely to be the "fully-quantized" models:
```
Comment (jeffxtang) [inline prototype_source/nnapi_mobilenetv2.rst:143]: it'd be great to show benchmark data here.. maybe in a future update.

File: prototype_source/nnapi_mobilenetv2.rst [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+Introduction
+------------
+
+This tutorial shows how to prepare a computer vision model to use
+`Android's Neural Networks API (NNAPI) <https://developer.android.com/ndk/guides/neuralnetworks>`_.
+NNAPI provides access to powerful and efficient computational cores
+on many modern Android devices.
+
+PyTorch's NNAPI is currently in the "prototype" phase and only supports
```
[the remark this answers is not in the document]
Comment (dreiss) [reply prototype_source/nnapi_mobilenetv2.rst:12]: Updating to reference the specific version that is known to work.

File: prototype_source/nnapi_mobilenetv2.rst [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+and expand our operator support over time.
+
+
+Model Preparation
+-----------------
+
+First, we must prepare our model to execute with NNAPI.
+This step runs on your training server or laptop.
+The key conversion function to call is
+``torch.backends._nnapi.prepare.convert_model_to_nnapi``,
```
[the remark this answers is not in the document]
Comment (dreiss) [reply prototype_source/nnapi_mobilenetv2.rst:23]: No.  This still needs to be written.

File: prototype_source/nnapi_mobilenetv2.rst [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+but some extra steps are required to ensure that
+the model is properly structured.
+Most notably, quantizing the model is required
+in order to run the model on certain hardware backends.
```
[the remark this answers is not in the document]
Comment (dreiss) [reply prototype_source/nnapi_mobilenetv2.rst:27]: Clarified that this refers to accelerators.

File: prototype_source/nnapi_mobilenetv2.rst [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+    import torch.backends._nnapi.prepare
+    import torchvision.models.quantization.mobilenet
+    from pathlib import Path
+
+
+    # This script supports 3 modes of quantization:
+    # - "none": Fully floating-point model.
+    # - "core": Quantize the core of the model, but wrap it a
+    #    quantizer/dequantizer pair, so the interface uses floating point.
+    # - "full": Quantize the model, and use quantize tensors
```
[the remark this answers is not in the document]
Comment (dreiss) [reply prototype_source/nnapi_mobilenetv2.rst:50]: Thanks.

File: prototype_source/nnapi_mobilenetv2.rst (hunk 1 above)
Comment (dreiss) [reply prototype_source/nnapi_mobilenetv2.rst:41]: Updated instructions to reference the specific known-working version.

File: prototype_source/nnapi_mobilenetv2.rst (hunk 2 above)
Comment (dreiss) [reply prototype_source/nnapi_mobilenetv2.rst:148]: Corrected.

File: prototype_source/nnapi_mobilenetv2.rst (hunk 3 above)
Comment (dreiss) [reply prototype_source/nnapi_mobilenetv2.rst:143]: Yeah.  I'll add this after the blog post goes live.

**Pull request merged**